tools/src/main/scala/scala/scalanative/codegen/CodeGen.scala

```diff
@@ -12,6 +12,13 @@ import scalanative.nir._
 
 object CodeGen {
 
+  private final class Platform(target: String) {
+    val isWindows = target.contains("indows")
```

Comment:
SBT which includes all the build and plugin contains some handy code to deal with Files and Paths. http://www.scala-sbt.org/0.13/docs/Paths.html  Note that normal File has a constructor that takes a `File(dir: String or File, file: String)` https://docs.oracle.com/javase/7/docs/api/java/io/File.html and there is also `java.nio.file` that other methods. Most of the String encoded paths not using File.separator code could be replaced.

This won't work for 3. above as you point out above.